spec/helpers/appstore_project_helper.rb

```diff
@@ -0,0 +1,214 @@
+          cc_process = project.deploy_process(
+            CC_PROCESS_ARCHIVE,
+            name: CC_PROCESS_NAME
+          )
+          cc_schedule = cc_process.create_schedule(
+            CC_SCHEDULE_CRON,
+            CC_GRAPH,
+            params: CC_PARAMS,
+            hidden_params: CC_SECURE_PARAMS,
+            state: 'DISABLED'
+          )
+          ruby_process = project.deploy_process(
+            RUBY_HELLO_WORLD_PROCESS_PATH,
+            name: RUBY_HELLO_WORLD_PROCESS_NAME
+          )
+          ruby_schedule = ruby_process.create_schedule(
+            cc_schedule,
+            'main.rb',
+            params: RUBY_PARAMS,
+            hidden_params: RUBY_SECURE_PARAMS,
+            state: 'DISABLED'
+          )
+
+          associate_output_stage ads
+          add_process = project.add.process
+          add_process.create_schedule(
+            ruby_schedule,
+            '',
+            dataload_datasets: [ DATASET_IDENTIFIER ],
+            de_synchronize_all: true,
+            state: 'DISABLED'
+          )
+        end
+      end
+
+      def create_metrics
+        GoodData.with_project @project.pid do |project|
+          metrics = JSON.parse(File.read(METRICS_FILE)).to_hash
+          metrics.each do |title, maql|
+            metric = project.add_measure maql, title: title
+            metric.identifier = "metric.#{title.to_s.downcase.gsub(/\s/, '.')}"
+            metric.add_tag('metric') if metric.identifier == PRODUCTION_TAGGED_METRIC
+            metric.save
+          end
+        end
+      end
+
+      def create_computed_attributes
+        update = GoodData::Model::ProjectBlueprint.build('update') do |project_builder|
+          project_builder.add_computed_attribute(
+            COMPUTED_ATTRIBUTE_ID,
+            title: 'My computed attribute',
+            metric: 'metric.customers.count',
+            attribute: 'attr.csv_policies.state',
+            buckets: [{ label: 'Small', highest_value: 1000 }, { label: 'Medium', highest_value: 2000 }, { label: 'Large' }]
+          )
+        end
+        new_blueprint = project.blueprint.merge(update)
+        unless new_blueprint.valid?
+          pp new_blueprint.validate
+          raise 'Cannot create computed attribute'
+        end
+        project.update_from_blueprint(new_blueprint)
+
+        project.attributes(COMPUTED_ATTRIBUTE_ID).drill_down(project.attributes('attr.csv_policies.customer'))
+      end
+
+      def create_reports
+        GoodData.with_project @project.pid do |p|
+          reports = JSON.parse(File.read(REPORTS_FILE)).to_hash['reports']
+          reports.each do |r|
+            report = p.add_report(
+              title: r['title'],
+              left: r['left'].map { |o| GoodData::MdObject[o] },
+              top: r['top'].map { |o| GoodData::MdObject[o] }
+            )
+            report_name = r['title'].to_s.downcase.gsub(/\s/, '.')
+            report.identifier = "report.#{report_name}"
```

Comment:
Refers to 'report' more than self (maybe move it to another class?) (FeatureEnvy)